types/applepayjs/index.d.ts

```diff
@@ -874,6 +874,46 @@ declare namespace ApplePayJS {
         tokenNotificationURL?: string;
     }
 
+    /**
+     * A dictionary that represents a request to set up a deferred payment, such as a hotel booking or a pre-order.
+     */
+    interface ApplePayDeferredPaymentRequest {
+        /**
+         * The localized billing agreement the framework displays to the user prior to payment authorization.
+         */
+        billingAgreement?: string | undefined;
+
+        /**
+         * A dictionary that contains details about the deferred payment.
+         */
+        deferredBilling: ApplePayLineItem;
+
+        /**
+         * [ISO 8601 formatted] The time zone at the destination location of the payment.
+         */
+        freeCancellationDate?: string | undefined;
```

Comment:
Is this a `string` and not a `Date`? The documentation uses `Date` not `DOMString` (which are then usually just mapped to `string`).

Reply:
Just to confirm, is there something yet to be fixed here? thanks 🙂

Reply:
I don't know, hence the question. The code appears to not match the documentation.

Reply:
The documentation on those properties in the code here talk about how the string is formatted, but the Apple documentation gives them as `Date`. Maybe a later version of ApplePay JS since those additions were added here originally changed the type from `string` to `Date`?

I suggest you investigate the behaviour in an actual integration on a device, because based solely on Apple's documentation I would expect this property to be a `Date`, so a user could do things like access the year directly rather than having to parse it first etc.

Reply:
Hmm got your point. I thought it could be a standard of this package, but it was not the case.
Just made a test using the ApplePay playground, and this is what I got:

```
    "freeCancellationDate": "2024-02-17T00:00:00.000Z",
    "freeCancellationDateTimeZone": "GMT",
```

Just made a new commit changing the `freeCancellationDate` to Date.

I can open a PR after this one updating also `recurringPaymentStartDate`, `recurringPaymentEndDate` and `deferredPaymentDate`. Or do you think it is worth it to have it already fixed in this same PR?

Reply:
I'm not sure how that shows which type the property is at runtime, as that appears to just be some JSON serialized data.

What do you get at runtime if you do something like this?

```javascript
let type = typeof foo.freeCancellationDate;
console.log(type);
```

For example, if I do this I get the following output:

```javascript
let date = new Date();
let dateString = "2024-02-17T00:00:00.000Z";

let typeDate = typeof date;
let typeString = typeof dateString;

console.log(typeDate); // "object"
console.log(typeString); // "string"
```

Reply:
Hey, in fact we do not receive that value, we just pass it to the `completeCouponCodeChange`.

I did the following tests:
- Sent a `new Date()` - it worked and showed the right value in the apple pay prompt
- Sent a `JSON.stringify(new Date())` - it also worked showing the right value in the prompt
- Sent 'teste', just to see how it would react to a invalid value, and the prompt does not even work.

Probably internally the `completeCouponCodeChange` get the value received and use a new Date() on it.
So Date and any string that work through a `new Date()` worked.